substrate-node/pallets/pallet-tfgrid/src/tests.rs

```diff
@@ -2322,6 +2320,31 @@ fn test_farming_policies_ordering_and_assignment() {
     })
 }
 
+#[test]
+fn test_node_gpu_status_update() {
+    ExternalityBuilder::build().execute_with(|| {
+        create_twin();
+        create_farm();
+        create_node();
+
+        assert_ok!(TfgridModule::set_node_gpu_status(
+            RuntimeOrigin::signed(alice()),
+            true,
+        ));
+
+        let status = TfgridModule::node_gpu_status(1);
+        assert_eq!(status, true);
+
+        assert_ok!(TfgridModule::set_node_gpu_status(
+            RuntimeOrigin::signed(alice()),
+            false,
+        ));
+
+        let status = TfgridModule::node_gpu_status(1);
+        assert_eq!(status, false);
+    });
+}
+
```

Comment:
I would just rename test:

`test_set_node_gpu_status_works()`

and  add 2 more tests:

`test_set_node_gpu_status_twin_not_exists_fails()`
`test_set_node_gpu_status_node_not_exists_fails()`

but this is personal preference ;)

Reply:
Would only add `test_set_node_gpu_status_node_not_exists_fails` because you cannot create a node without creating a twin